sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -436,23 +421,19 @@ public void processElement(ProcessContext context) throws Exception {
       @FinishBundle
       public void finishBundle(Context context) throws Exception {
         for (T record : batch) {
-          try {
-            PreparedStatement statement = connection.prepareStatement(query);
-            preparedStatementSetter.setParameters(record, statement);
-            try {
-              statement.executeUpdate();
-            } finally {
-              statement.close();
-            }
-          } catch (Exception e) {
-            LOGGER.error("Can't insert into database", e);
-          }
+          preparedStatementSetter.setParameters(record, preparedStatement);
+          preparedStatement.executeUpdate();
```

Comment:
Seems reasonable. Also set batchCount back to 0 after executing the batch. And make sure to commit the batch as a single transaction, for efficiency (http://stackoverflow.com/questions/6892105/bulk-insert-in-java-using-prepared-statements-batch-update recommends not using autocommit when using this pattern)